test/extensions/transport_sockets/http_11_proxy/connect_test.cc

```diff
@@ -560,6 +593,94 @@ TEST(ParseTest, ContentLengthZeroHttp11) {
+
+  // Test with IPv4 address with port
+  EXPECT_EQ("CONNECT 192.168.1.1:8080 HTTP/1.1\r\nHost: 192.168.1.1:8080\r\n\r\n",
+            UpstreamHttp11ConnectSocket::formatConnectRequest("192.168.1.1:8080"));
+
+  // Test with IPv6 address without port
+  EXPECT_EQ("CONNECT [2001:db8::1] HTTP/1.1\r\nHost: [2001:db8::1]\r\n\r\n",
+            UpstreamHttp11ConnectSocket::formatConnectRequest("[2001:db8::1]"));
+
+  // Test with IPv6 address with port
+  EXPECT_EQ("CONNECT [2001:db8::1]:443 HTTP/1.1\r\nHost: [2001:db8::1]:443\r\n\r\n",
+            UpstreamHttp11ConnectSocket::formatConnectRequest("[2001:db8::1]:443"));
+}
+
+// Test runtime guard for legacy behavior with transport socket options
+TEST_P(Http11ConnectTest, RuntimeGuardLegacyBehaviorTransportSocketOpts) {
+  TestScopedRuntime scoped_runtime;
+  scoped_runtime.mergeValues(
+      {{"envoy.reloadable_features.http_11_proxy_connect_legacy_format", "true"}});
+
+  initialize();
+
+  Buffer::OwnedImpl msg("initial data");
+  Buffer::OwnedImpl expected_legacy_data{"CONNECT www.foo.com:443 HTTP/1.1\r\n\r\n"};
+
+  EXPECT_CALL(io_handle_, write(BufferStringEqual(expected_legacy_data.toString())))
+      .WillOnce(Invoke([&](Buffer::Instance& buffer) {
+        auto length = buffer.length();
+        buffer.drain(length);
+        return Api::IoCallUint64Result(length, Api::IoError::none());
+      }));
+
+  Network::IoResult rc1 = connect_socket_->doWrite(msg, false);
+  EXPECT_EQ(expected_legacy_data.length(), rc1.bytes_processed_);
+
+  // Second write should be buffered (returns 0) until onConnected() is called
+  Network::IoResult rc2 = connect_socket_->doWrite(msg, false);
+  EXPECT_EQ(0, rc2.bytes_processed_);
+
+  EXPECT_CALL(*inner_socket_, onConnected());
+  connect_socket_->onConnected();
+}
+
+// Test runtime guard for legacy behavior with metadata proxy address
+TEST_P(Http11ConnectTest, RuntimeGuardLegacyBehaviorEndpointMetadata) {
+  TestScopedRuntime scoped_runtime;
+  scoped_runtime.mergeValues(
+      {{"envoy.reloadable_features.http_11_proxy_connect_legacy_format", "true"}});
+
+  initializeWithMetadataProxyAddr();
+
+  Buffer::OwnedImpl msg("initial data");
+  // Use static string to avoid dynamic allocation issues in lambda capture
+  const std::string expected_connect_string = absl::StrCat(
+      "CONNECT ", Network::Test::getLoopbackAddressUrlString(GetParam()), ":1234 HTTP/1.1\r\n\r\n");
+  Buffer::OwnedImpl expected_legacy_data{expected_connect_string};
+
+  EXPECT_CALL(io_handle_, write(BufferStringEqual(expected_legacy_data.toString())))
+      .WillOnce(Invoke([&](Buffer::Instance& buffer) {
+        auto length = buffer.length();
+        buffer.drain(length);
+        return Api::IoCallUint64Result(length, Api::IoError::none());
+      }));
+
+  Network::IoResult rc1 = connect_socket_->doWrite(msg, false);
+  EXPECT_EQ(expected_legacy_data.length(), rc1.bytes_processed_);
+
+  // Second write should be buffered (returns 0) until onConnected() is called
```

Comment:
You might as well verify this after calling onConnected().

Reply:
Please see if the last change is what you have in mind for this test.